lightning/src/ln/channelmanager.rs

```diff
@@ -628,6 +628,13 @@ pub(super) struct PeerState<Signer: ChannelSigner> {
 	/// Messages to send to the peer - pushed to in the same lock that they are generated in (except
 	/// for broadcast messages, where ordering isn't as strict).
 	pub(super) pending_msg_events: Vec<MessageSendEvent>,
+	/// Map from Channel IDs to pending [`ChannelMonitorUpdate`]s which have been passed to the
+	/// user but which have not yet completed.
+	///
+	/// Note that the channel may no longer exist. For example if the channel was closed but we
+	/// later needed to claim an HTLC which is pending on-chain, we may generate a monitor update
+	/// for a missing channel.
+	in_flight_monitor_updates: BTreeMap<OutPoint, Vec<ChannelMonitorUpdate>>,
```

Comment:
I'm sort of confused on when/how something like this would occur? Would it be like we got a preimage for an HTLC after a channel was closed?

Reply:
How would that happen though? I'm looking at `ChannelManager::internal_hupdate_fulfill_htlc` and we error if the channel doesn't exist?

Reply:
Hmm, is that upstream channel not going to do the same thing when you forward it the preimage?

Reply:
Ahhh okay that makes sense, thanks!